client/state/sites/plans/selectors/is-site-on-woo-express.ts

```diff
@@ -16,16 +17,19 @@ import type { AppState } from 'calypso/types';
  */
 export default function isSiteOnWooExpress( state: AppState, siteId: number ) {
 	const currentPlan = getCurrentPlan( state, siteId );
+	const site = getSite( state, siteId );
 	const wooExpressPlans = [
 		PLAN_WOOEXPRESS_MEDIUM,
 		PLAN_WOOEXPRESS_SMALL,
 		PLAN_WOOEXPRESS_MEDIUM_MONTHLY,
 		PLAN_WOOEXPRESS_SMALL_MONTHLY,
 	];
 
-	if ( ! currentPlan ) {
+	const productSlug = currentPlan?.productSlug || site?.plan?.product_slug;
```

Comment:
It seems `site` data also has the current plan slug. This is necessary for better UX since `site` data is already available when the themes screen is loaded. Otherwise, we'll have some upsell component flicker in-and-out and the themes list not loading properly since hidden filter is added after its first load.